src/OpenLoco/src/Vehicles/VehicleBogie.cpp

```diff
@@ -119,6 +104,164 @@ namespace OpenLoco::Vehicles
         }
     }
 
+    // To replace several instances of the same exact code in
+    // VehicleBogie::updateSegmentCrashed(). Returns true if the
+    // vehicle crashed in the process.
+    bool VehicleBogie::rotateAndExplodeIfNotAlreadyExploded()
+    {
+        // update sprite yaw
+        this->spriteYaw = (this->spriteYaw + 4) & 0x3F;
+
+        // explode, if we haven't already
+        if (!this->hasVehicleFlags(VehicleFlags::unk_5))
+        {
+            this->explodeComponent();
+            this->vehicleFlags |= VehicleFlags::unk_5;
+            return true;
+        }
+        else
+        {
+            return false;
+        }
+    }
+
+    // 0x004AA68E
+    void VehicleBogie::updateSegmentCrashed()
+    {
+        _vehicleUpdate_frontBogie = _vehicleUpdate_backBogie;
+        _vehicleUpdate_backBogie = X86Pointer(this);
+
+        uint32_t speed = this->var_5A & 0x7FFFFFFF;
+        uint32_t var_5A_msb_flag_temp = this->var_5A & 0x80000000;
+        speed = speed - (speed / 64);
+        if (speed <= 8192)
+        {
+            speed = 0;
+        }
+
+        this->var_5A = speed | var_5A_msb_flag_temp;
+        _vehicleUpdate_var_113612C = speed / 128;
```

Comment:
i would create a temporary var `const auto distanceTravelled = speed.getRaw() / 128; // distance travelled this tick`